staging/src/github.com/kubeedge/viaduct/pkg/packer/reader.go

```diff
@@ -27,25 +31,43 @@ func (r *Reader) Read() ([]byte, error) {
 	}
 
 	headerBuffer := make([]byte, HeaderSize)
-	_, err := io.ReadFull(r.reader, headerBuffer)
+	n, err := io.ReadFull(r.reader, headerBuffer)
 	if err != nil {
 		if err != io.EOF {
-			klog.Error("failed to read package header from buffer")
+			klog.Error("failed to read package header from buffer", err)
 		}
 		return nil, err
 	}
+	if HeaderSize != n {
+		err := fmt.Errorf("failed to read package header: invalid header, length want %d, but got %d", HeaderSize, n)
+		klog.Error(err)
+		return nil, err
+	}
```

Comment:
`io.ReadFull` will return an error if fewer bytes were read, so i dont think this `if` statement is neccessary when we have already decided whether `err` is nil or not.